### DISCUSSION TITLE
How to get Profile Pair Extraordinaire

### DISCUSSION THREAD

**Discussion by rodukov** [community/community#20467] [Other Feature Feedback, Questions, & Ideas] [answered]
📗 **Recently there are new achievements and I'm wondering how to get the Pair Extraordinaire**
_Watch all achievements: https://github.com/Schweinepriester/github-profile-achievements_
<img width="148" alt="pair-extraordinaire-default" src="https://user-images.githubusercontent.com/101352977/178841186-98adb2c7-3c39-4e69-8251-09891cbe1983.png">

**Comment by radar-blip** [comment]
You must co-author commits on merged pull requests. 👍🏼

**Reply by rodukov** [reply]
I don't understand what co-author means. Give **a detailed** and understandable answer and I will choose it as the main one.

**Reply by radar-blip** [reply]
What devkennyy said. You, and one or more members must authorize a pull request.

**Comment by devkennyy** [comment]
You must be one of two (or more) authors on a pull request. View [this link ](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors)for more info.

**Comment by treblenalto** [comment, marked as answer]
This is one way to do it
1. create a branch and commit changes
3. get a collaborator to commit changes on the same branch
4. create a pull request 
2. merge to default branch using _squash and merge_

**Reply by sibeux** [reply]
> This is one way to do it
> 
> 1. create a branch and commit changes
> 2. get a collaborator to commit changes on the same branch
> 3. create a pull request
> 4. merge to default branch

i still cant get the achievement, can you explain in more detail?

**Reply by thenithinbalaji** [reply]
@rodukov did this work? If it didn't try mine and let me know!

**Reply by BEPb** [reply]
Thank you @Taehee-K its work

**Reply by xmas7** [reply]
awesome

**Reply by QuantumFluxx** [reply]
thanks

**Reply by IFox99** [reply]
> This is one way to do it
> 
> 1. create a branch and commit changes
> 2. get a collaborator to commit changes on the same branch
> 3. create a pull request
> 4. merge to default branch using _squash and merge_

Your way is wrong don't believe this man 
he wants to keep all the achievements to himself 
he's selfish 
don't squash the pull request just regular merge it 
and thank you for wasting my time 
wait no, no thank you

**Reply by Sandro642** [reply]
thanks

**Reply by Sandro642** [reply]
what is it squash ?

**Reply by MudasirHussain72** [reply]
In GitHub, "squash" refers to combining multiple commits into a single commit.

**Reply by adeemaamir240** [reply]
❤💖

**Reply by Luisa-m17** [reply]
❤❤

**Reply by Tvenus** [reply]
💖💖💖

**Reply by Ramedon1** [reply]
💖💖💖
❤❤

**Reply by de-pesh** [reply]
I dont know why it's not working for me 😪

1) I created a repo and added a collaborator
2) I created a branch  and commit changes in this branch
3) The collaborator also made changes in the same branch
4) I created a pull request
5) I merged the PR 

Did I do something wrong?

**Reply by Tvenus** [reply]
Add collaborators and commit from GitHub Desktop.

**Reply by de-pesh** [reply]
Thanks it worked

**Reply by Tvenus** [reply]
okay

**Reply by HimashaNethmini** [reply]
> I dont know why it's not working for me 😪
> 
> 1. I created a repo and added a collaborator
> 2. I created a branch  and commit changes in this branch
> 3. The collaborator also made changes in the same branch
> 4. I created a pull request
> 5. I merged the PR
> 
> Did I do something wrong?

did you install github dekstop. you have to do that using github desktop

**Reply by de-pesh** [reply]
> > I dont know why it's not working for me 😪
> > 
> > 1. I created a repo and added a collaborator
> > 2. I created a branch  and commit changes in this branch
> > 3. The collaborator also made changes in the same branch
> > 4. I created a pull request
> > 5. I merged the PR
> > 
> > Did I do something wrong?
> 
> did you install github dekstop. you have to do that using github desktop

yeah miss, i did it and it worked
thanks anyways

**Comment by diegoleondev** [comment]
👋 Hi @lgrachov, These are all the achievements so far:

## Achievements

| Title | Badge | Earnable? | Earned by |
| --- | --- | --- | --- |
Pair Extraordinaire | ![Pair Extraordinaire Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/pair-extraordinaire-default.png) | ✔️ | Coauthored in a merged pull request
Quickdraw | ![Quickdraw Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/quickdraw-default.png) | ✔️ | Closed an issue or a pull request within 5 min of opening
Starstruck | ![Starstruck Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/starstruck-default.png) | ✔️ | Created a repository that has 16 stars
||| 
Galaxy Brain | ![Galaxy Brain Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/galaxy-brain-default.png) | ✔️ | 2 accepted answers
Pull Shark | ![Pull Shark Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/pull-shark-default.png) | ✔️ | 2 pull requests merged
YOLO | ![YOLO Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/yolo-default.png) | ✔️ | Merged a pull request without code review
||| 
Arctic Code Vault Contributor | ![Arctic Code Vault Contributor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/arctic-code-vault-contributor-default.png) | ❌ | Contributed code to repositories in the [2020 GitHub Archive Program](https://archiveprogram.github.com/)
Public Sponsor | ![GitHub Sponsor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/public-sponsor-default.png) | ✔️ | Sponsoring open source work via [GitHub Sponsors](https://github.com/sponsors)
Mars 2020 Contributor | ![Mars 2020 Helicopter Contributor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/mars-2020-contributor-default.png) | ❌ | Contributed code to repositories used in the [Mars 2020 Helicopter Mission](https://github.com/readme/nasa-ingenuity-helicopter)

You can supplement the information in the following github article: https://github.com/Schweinepriester/github-profile-achievements/blob/main/README.md

>**Note**
>Please **Mark as answer** if it was helpful.

**Reply by ikramulkayes** [reply]
Are they going to add more achievements?

**Reply by thenithinbalaji** [reply]
> Are they going to add more achievements?

maybe, coz Starstruck, Quickdraw and Pair Extraordinaire are newly added ones

**Reply by qxprakash** [reply]
> 👋 Hi @lgrachov, These are all the achievements so far:
> 
> ## Achievements
> Title	Badge	Earnable?	Earned by
> Pair Extraordinaire	![Pair Extraordinaire Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/pair-extraordinaire-default.png)	✔️	Coauthored in a merged pull request
> Quickdraw	![Quickdraw Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/quickdraw-default.png)	✔️	Closed an issue or a pull request within 5 min of opening
> Starstruck	![Starstruck Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/starstruck-default.png)	✔️	Created a repository that has 16 stars
> Galaxy Brain	![Galaxy Brain Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/galaxy-brain-default.png)	✔️	2 accepted answers
> Pull Shark	![Pull Shark Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/pull-shark-default.png)	✔️	2 pull requests merged
> YOLO	![YOLO Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/yolo-default.png)	✔️	Merged a pull request without code review
> Arctic Code Vault Contributor	![Arctic Code Vault Contributor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/arctic-code-vault-contributor-default.png)	❌	Contributed code to repositories in the [2020 GitHub Archive Program](https://archiveprogram.github.com/)
> Public Sponsor	![GitHub Sponsor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/public-sponsor-default.png)	✔️	Sponsoring open source work via [GitHub Sponsors](https://github.com/sponsors)
> Mars 2020 Contributor	![Mars 2020 Helicopter Contributor Achievement Badge](https://github.com/Schweinepriester/github-profile-achievements/raw/main/images/mars-2020-contributor-default.png)	❌	Contributed code to repositories used in the [Mars 2020 Helicopter Mission](https://github.com/readme/nasa-ingenuity-helicopter)
> You can supplement the information in the following github article: https://github.com/Schweinepriester/github-profile-achievements/blob/main/README.md
> 
> > Note
> > Please **Mark as answer** if it was helpful.

Thanks for sharing this @dalldev

**Reply by Golgrax** [reply]
i think the quickdraw can't be collect now.

**Comment by tskxz** [comment]
I strongly recommend that you use Github Desktop to be easier to add a co-author 
![image](https://user-images.githubusercontent.com/47666341/179217800-0a0aa730-b07a-4c28-ab83-19a9581079fb.png)

**Reply by BEPb** [reply]
Thank you @8fn its work!!! Great! You my a co-author

**Reply by Tvenus** [reply]
💖💖💖

**Comment by BEPb** [comment]
It is work:
1. create a branch and commit changes
2. get a collaborator to commit changes on the same branch
3. create a pull request
4. merge to default branch

**Reply by ITProggeruz** [reply]
yes

**Reply by Prakash4844** [reply]
Hey check this out. https://github.com/Prakash4844/Get-Pair-Extraordinaire

**Comment by muath-ye** [comment]
It's so much easy

1- Create a branch and commit changes as following:

By using [Github Desktop](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-using-github-desktop)

![image](https://docs.github.com/assets/cb-1355315/images/help/desktop/co-authors-demo-hq.gif)

Or by [Command line](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line)

```bat
$ git commit -m "Refactor usability tests.
>
>
Co-authored-by: name <name@example.com>
Co-authored-by: another-name <another-name@example.com>"
```

2- Create a pull request

3- Merge to default branch

**Comment by coreymtaylor** [comment]
https://github.com/Schweinepriester/github-profile-achievements

**Comment by weihaox** [comment]
The following steps can be taken on **Github Desktop** to get the pair-extraordinaire badge:

1. find a repo, e.g., your own repository, and make some changes there;
2. select the changed file and add co-authors (left-bottom of the descriptions);
3. commit and wait for the pull request to be merged (when using your own repository, you need to create a new branch by choosing _Branch -> New Branch_ and merge your pull request on Github website);
4. congratulations, you have been awarded the pair-extraordinaire badge.

**Reply by Tvenus** [reply]
💖💖💖

**Comment by aimonext** [comment]
create a branch and commit changes
get a collaborator to commit changes on the same branch
create a pull request
merge to the default branch using squash and merge

**Comment by nsafarova** [comment]
you should be Coauthored in a merged pull request.

**Comment by SUBHADIPMAITI-DEV** [comment]
1. create a branch and commit changes
2. get a collaborator to commit changes on the same branch
3. create a pull request
4. merge to the default branch using squash and merge

**Comment by saket61195** [comment]
without github desktop application you will not get **Pair Extraordinaire** badge
currently it is available on windows and mac so download github app and follows these steps: 
make sure you added  Collaborator into you repo(in setting option you will get)
1. open github app
2. clone repo where you want to add co author
3. make a new branch
4. open that branch into vscode or any editor
5. make changes into any files and save it
6. you will see into github app some changes shown already
7.  add **co-author** commit that cahnges and push
8. pull request and merge using squash and merge
9. you and your co-author with get **Pair Extraordinaire** badge

**Comment by Prakash4844** [comment]
Easiest way is to just use this repo that automates the process: https://github.com/Prakash4844/Get-Pair-Extraordinaire

**Comment by BhagyaAmarasinghe** [comment]
Pair extraordinaire can be achieved by co-authoring a commit. You have to include `Co-authored-by: NAME <NAME@EXAMPLE.COM>` in your commit message. For exmple:

```
$ git commit -m "Refactor usability tests.
>
>
Co-authored-by: NAME <NAME@EXAMPLE.COM>
Co-authored-by: ANOTHER-NAME <ANOTHER-NAME@EXAMPLE.COM>"
```

Github documentation for co-authoring commits:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Comment by knownothing10** [comment]
thanks for all answers

**Reply by Tvenus** [reply]
💖💖💖
